### ISSUE TITLE
The main difference between mean_teacher and swa

### ISSUE THREAD

**Issue by SpinachR** [benathi/fastswa-semi-sup#2]
According to my understanding, the mean_teacher model attempts to average (exponential mean average) the weights for every training steps and including a consistency loss. While swa only attempts to average the weights for different epochs after the cyclic changes of the learning rate. 

AM I correct?
